dashboard/app/controllers/featured_projects_controller.rb

```diff
@@ -1,23 +1,35 @@
 class FeaturedProjectsController < ApplicationController
+  # The first time a project is featured.
   def create
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
+    if current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
       _, channel_id = storage_decrypt_channel_id(featured_project_params[:project_id])
-      @featured_project = FeaturedProject.create({storage_app_id: channel_id, who_featured_user_id: current_user.id})
+      @featured_project = FeaturedProject.create({storage_app_id: channel_id, featured_at: DateTime.now, unfeatured_at: nil})
+      puts "featured project in create #{@featured_project}"
       @featured_project.save!
     end
   end
 
-  def destroy_by_project_id
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(params[:project_id])
-      @featured_project = FeaturedProject.find_by_storage_app_id(channel_id)
-      @featured_project.destroy
+  def unfeature
+    if current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
+      _, channel_id = storage_decrypt_channel_id(featured_project_params[:project_id])
+      puts "channel_id > #{channel_id}"
+      @featured_project = FeaturedProject.find_by storage_app_id: channel_id
+      @featured_project.update_attribute(:unfeatured_at, DateTime.now)
+    end
```

Comment:
Same as above. Unless we actually want this to always return success even for bad permissions, we should use `CanCan` for proper auth. One `authorize_resource` call at the top of the class can take the place of all these specific `current_user.try(:permission?...` calls once the ability is set up.

This has another (presumably unintentional) `puts`

Finally, what about bad storage app ids? What if someone passes a bad URL? Presumably we'd want to 404 rather than blow up with a 500 internal server error, which is what will happen now after line 16 returns nil and line 17 attempts to call `update_attribute!` on nil. This too can be handled by CanCan with a `load_resource` or `load_and_authorize_resource`. I realize that the `project_id` is not the same as `featured_project_id`, but CanCan(Can) can still work with this. There are a bunch of helpful params to expand beyond the default for resources. See http://www.rubydoc.info/github/CanCanCommunity/cancancan/CanCan/ControllerAdditions/ClassMethods#load_resource-instance_method.

 Let me know if you want some help setting it up.